library/Ivoz/Provider/Domain/Model/WebPortal/WebPortal.php

```diff
@@ -36,6 +38,19 @@ protected function sanitizeValues(): void
             );
             throw new \DomainException($errorMsg);
         }
+
+        if (!$this->isGeneric()) {
+            $isUserUrlType = $this->getUrlType() === self::URLTYPE_USER;
+            if ($isUserUrlType && !$this->getCompany()?->isVpbx()) {
+                throw new \DomainException("Company must be vpbx in user urls");
+            }
+
+            $companyBrandId = $this->getCompany()?->getBrand()->getId();
+            $brandId = $this->getBrand()?->getId();
+            if ($companyBrandId !== $brandId) {
+                throw new \DomainException("Company brands must be identical to Brand");
+            }
+        }
```

Comment:
```suggestion
        // $company = $this->getCompany();
        // if(is_null($company)){
        //    return;
        // }
        
        // Check if the URL type is a user URL
        $isUserUrlType = $this->getUrlType() === self::URLTYPE_USER;

        // If the URL is a user URL and the company is not a VPBX, throw an exception
        if ($isUserUrlType) {
            $company = $this->getCompany();
            if ($company === null || !$company->isVpbx()) {
                throw new \DomainException("Company must be VPBX in user URLs");
            }
        }

        // Retrieve company brand ID and brand ID
        $company = $this->getCompany();
        $companyBrandId = $company 
            ? $company->getBrand()->getId() 
            : null;
        $brandId = $$this->getBrand()?->getId()

        // If the company brand ID does not match the brand ID, throw an exception
        if ($companyBrandId !== $brandId) {
            throw new \DomainException("Company brands must be identical to Brand");
        }
        
```

Reply:
It is not compulsory for a WebPortal to have a company. That's why I introduced the isGeneric() method. Generic is used in front for a WebPortal not binded to a company.

If a portal have not company then the new rules don’t apply at all.